### PR TITLE
Hompage Functionality and Reamining playlist setup

### PR DESCRIPTION
**Pull request by shobit1337**
What changes I have made:
- The homepage now shows Videos based on categories.
- Created Single Playlist Page.
- Add to playlist and Remove from playlist functionality added.
- minor UI improvements.
-

### CODE REVIEW

**Review by ScoobyDubeyDo — APPROVED** [review]
code looks clean

File: src/Routes.jsx [hunk 1]
```diff
@@ -5,49 +5,51 @@ import {
...
-              <Route
-                path='watch-later'
-                element={<ListingPage title={'Your watch later list'} />}
-              />
+        <Route path='/' element={<App />}>
+          <Route index element={<HomePage />} />
+
+          <Route path='browse' element={<ListingPage />} />
+
+          <Route element={<PrivateRoute />}>
```
Comment (xsaurabhsuthar) [inline src/Routes.jsx:30]: Good use of Route with Outlet.

File: src/components/CreatePlaylist/CreatePlaylist.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, { useRef } from 'react';
+import { useAuth } from '../../context/auth-context';
+import { usePlaylists } from '../../context/playlist-context';
+import { createPlaylist } from '../../store/playlist/actions';
+
+const CreatePlaylist = ({ closeHandler }) => {
+  const title = useRef('');
```
Comment (xsaurabhsuthar) [inline src/components/CreatePlaylist/CreatePlaylist.jsx:7]: avoid using uncontrolled input state.

**Review by Launchpad5682 — APPROVED** [review]
I've shared some suggestions. Otherwise, everything looks good. Good to merge 🚀🚀🚀🚀

File: src/backend/controllers/PlaylistController.js [hunk 3]
```diff
@@ -141,10 +141,12 @@ export const removeVideoFromPlaylistHandler = function (schema, request) {
     const playlistId = request.params.playlistId;
     const videoId = request.params.videoId;
     let playlist = user.playlists.find((item) => item._id === playlistId);
```
Comment (rutvikpumak) [inline src/backend/controllers/PlaylistController.js:143]: use const

File: src/components/CreatePlaylist/CreatePlaylist.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  const description = useRef('');
+  const { user } = useAuth();
+  const { dispatchPlaylists } = usePlaylists();
+
+  const createPlaylistHandler = async (e) => {
+    e.preventDefault();
+    let playlistTitle = title.current.value;
```
Comment (rutvikpumak) [inline src/components/CreatePlaylist/CreatePlaylist.jsx:14]: use const

File: src/pages/HomePage/HomePage.jsx [hunk 5]
```diff
@@ -1,15 +1,48 @@
...
     <div>
-      <h3 className='section-title'>Home of Reelflix!</h3>
-      <ListSection title='Latest Release' />
-      <ListSection title='Most Popular' />
-      <ListSection title='Thriller' />
-      <ListSection title='Romance' />
+      <h3 className='section-title'>Home of Reelflix! 🍿🕶</h3>
+      {videoList.length && (
+        <>
+          {' '}
```
Comment (rutvikpumak) [inline src/pages/HomePage/HomePage.jsx:24]: remove this

**Review by rutvikpumak — APPROVED** [review]
Suggest some changes, use most of the const until and unless you won't reassign some value to it.
The code looks well structured and clean.
make changes and you are good to to merge

**Pull request merged**